docs/DeveloperGuide.md

```diff
@@ -5,9 +5,251 @@
+7. After the loop ends, if the `isFound` variable is still false, the `noItemFound` method of the Ui class is called to notify that no item has been found containing the word
+
+### List Command
+The following is a class diagram of the FindCommand and its relevant dependencies<br>
+![ListCommandClass](uml-diagrams/ListCommandClass.png)
+
+The `ListCommand` class implements the `Command` interface and is responsible for printing out a list of items in the inventory to the output.
+A ListCommand instance is created by the `parseListCommand` method called by Parser, which parses the user input to determine how the list should be printed out.
+
+#### Dependencies
+- `Item`: For getting the comparator needed to sort the list of items
+- `Inventory`: For getting the list of items in the inventory
+- `Ui`: To print the list of items in the inventory to the output
+
+The 5 parameters in the constructor `hasQuantity`, `hasPrice`, `firstParam`, `sortBy`, `isReverse` are used to determine how the list should be printed out.
+- `hasQuantity`
+    - True if the user has input the quantity flag `q/`, false otherwise
+    - Quantity will be printed to the output if true
+- `hasPrice`
+    - True if the user has input the price flag `p/`, false otherwise
+    - Price will be printed to the output if true
+- `firstParam`
+    - Can only take 3 possible values `"q"`,`"p"` or `""`. 
+    - `"q"` if the quantity flag comes before the price flag.
+    - `"p"` if the price flag comes before the quantity flag
+    - If both quantity and price flag do not exist, then `""` would be the default value (this variable would no longer be used in the execution of `ListCommand`)
+    - Quantity will be printed before price if `"q"`
+    - Price will be printed before quantity if `"p"`
+- `sortBy`
+    - Can only take 3 possible values `"q"`,`"p"` or `""`
+    - `"q"` if the user has input the sort by quantity flag `sq/`
+    - `"p"` if the user has input the sort by price flag `sp/`
+    - `""` would be the default value if the user did not input any sorting flag
+    - If multiple sorting flags are input by the user, `sortBy` will take the value corresponding to the first sorting flag
+- `isReverse`
+    - True if the user has input the reverse flag `r/`, false otherwise.
+
+There are 6 sorting modes in total
+1. `sortBy == ""` and `isReverse == false`: Alphabetical ascending (e.g. A-Z)
+2. `sortBy == ""` and `isReverse == true`: Alphabetical descending (e.g. Z-A)
+3. `sortBy == "q"` and `isReverse == false`: Quantity ascending
+4. `sortBy == "q"` and `isReverse == true`: Quantity descending
+5. `sortBy == "p"` and `isReverse == false`: Price ascending
+6. `sortBy == "p"` and `isReverse == true`: Price descending
+
+The following sequence diagram shows the execution of a ListCommand<br>  
+![ListCommandSequence](uml-diagrams/ListCommandSequence.png)
```

Comment:
After calling a constructor, `new ArrayList<>...`, you can mark X in the end of the diagram to show that it is deleted